### PR TITLE
Support twitter previews in Draftail embeds. Fixes #5176

### LINKED ISSUE #5176: Support twitter previews in Draftail embeds
### Issue Summary
Draftail's EmbedBlock have troubles displaying tweets' informations (thumb / title)
  
### Steps to Reproduce
1. Start a new project with `wagtail start myproject`
2. Add a RichTextField to an example model
3. Add a new content for this example model 
4. Click embed block in draftail toolbar
5. Add a tweet `https://twitter.com/libe/status/1110520362930974721?s=20` 

Expectations: get a sumary (title, description, thumbnail or any other relevant information) displaying in the textarea
Result: Nothing except the generic look of embeds in the editor. 
  
As discussed on slack with @thibaudcolas , this comes from here: https://github.com/wagtail/wagtail/blob/b1fab7084719ddbb88b9581567660930a6c082b4/client/src/components/Draftail/blocks/EmbedBlock.js#L13-L27  

Also, as said on slack, on good generic workaround to this case (and other future cases like this one) could be to display embed source in the preview. 

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details
* Python version: 3.7.2
* Django version: 2.1.7
* Wagtail version: 2.4

### PR DESCRIPTION
**Pull request by imansyed000**
Fixes #5176
Tested on Chrome and Safari. 
Added new tests

image

If no thumbnails available via oembed, returns 'Provider name - Author name'

### CODE REVIEW

File: client/src/components/Draftail/blocks/EmbedBlock.js [hunk 1]
```diff
@@ -10,26 +10,40 @@ import MediaBlock from '../blocks/MediaBlock';
  */
 const EmbedBlock = props => {
   const { entity, onRemoveEntity } = props.blockProps;
-  const { url, title, thumbnail } = entity.getData();
+  const { url, title, thumbnail, providerName, authorName } = entity.getData();
+
+  if (thumbnail) {
```
Comment (thibaudcolas) [inline client/src/components/Draftail/blocks/EmbedBlock.js]: Could you turn this and the other case around? I think I’d rather have `!thumbnail` as a special case, then keep the `return` at the end of the function be the "has thumbnail" scenario.

File: client/src/components/Draftail/blocks/EmbedBlock.test.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ describe('EmbedBlock', () => {
                 url: 'http://www.example.com/',
                 title: 'Test title',
                 thumbnail: 'http://www.example.com/example.png',
+                authorName: 'Example author',
+                providerName: 'Example provider',
```
Comment (thibaudcolas) [inline client/src/components/Draftail/blocks/EmbedBlock.test.js]: Could you add a separate case for this test suite that specifically tests what this component renders when there is data, but not a thumbnail?

File: client/src/components/Draftail/blocks/EmbedBlock.js (hunk 1 above)
Comment (thibaudcolas) [inline client/src/components/Draftail/blocks/EmbedBlock.js]: Could you turn this and the other case around? I think I’d rather have `!thumbnail` as a special case, then keep the `return` at the end of the function be the "has thumbnail" scenario.

File: client/src/components/Draftail/blocks/EmbedBlock.js [hunk 3]
```diff
@@ -10,26 +10,40 @@ import MediaBlock from '../blocks/MediaBlock';
...
-          title={url}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          {title}
-        </a>
-      ) : null}
-
-      <button className="button button-secondary no Tooltip__button" onClick={onRemoveEntity}>
+    <strong>
```
Comment (thibaudcolas) [inline client/src/components/Draftail/blocks/EmbedBlock.js]: I think I would still like this case to look visually similar to the `MediaBlock` component, perhaps overlaying the `providerName` & `authorName` text you have here on top of the existing block visual. Could you add a prop to `MediaBlock` for that, and then use the block with that extra prop?

File: client/src/components/Draftail/blocks/EmbedBlock.js [hunk 4]
```diff
@@ -10,26 +10,40 @@ import MediaBlock from '../blocks/MediaBlock';
...
+      <a href={url} target={'_blank'}>
+        {providerName + ' - ' + authorName + '  '}
```
Comment (thibaudcolas) [inline client/src/components/Draftail/blocks/EmbedBlock.js]: Should be written with ES6 template strings.

File: client/src/components/Draftail/blocks/EmbedBlock.js [hunk 5]
```diff
@@ -10,26 +10,40 @@ import MediaBlock from '../blocks/MediaBlock';
...
+      </a>
+
+      <button className="button button-secondary no Tooltip__button" onClick={onRemoveEntity} >
```
Comment (thibaudcolas) [inline client/src/components/Draftail/blocks/EmbedBlock.js]: I think this should stay within the tooltip, so the block’s UI isn’t distracting while entering content.

File: client/src/components/Draftail/blocks/EmbedBlock.js (hunk 3 above)
Comment (imansyed000) [reply client/src/components/Draftail/blocks/EmbedBlock.js]: Applied your suggestions. Don't know how to approach it in terms of styling but we can discuss when you're back :)

Comment (imansyed000) [reply client/src/components/Draftail/blocks/EmbedBlock.js]: Preview looks like this

Screenshot 2019-08-16 at 09 49 55

Comment (imansyed000) [reply client/src/components/Draftail/blocks/EmbedBlock.js]: Update: Just noticed this doesn't work on Chrome

**Comment by thibaudcolas** [comment]
Closing this in favour of #5545.

**Pull request closed without merging**